gvm-libs-22.4.0 fails to build without paho-mqtt

Basically this has already been reported in the past https://github.com/greenbone/gvm-libs/issues/632 but not addressed, configure succeeds

```
[...]
-- Looking for paho-mqtt3c ... LIBPAHO-NOTFOUND
-- libpaho-mqtt3c is required for MQTTv5 support.
[...]
```

but build fails with:

```
[...]
[ 56%] Building C object util/CMakeFiles/gvm_util_shared.dir/mqtt.c.o
cd /var/tmp/paludis/build/net-analyzer-gvm-libs-22.4.0/work/build/util && /usr/bin/x86_64-pc-linux-gnu-cc -DGVM_LIBS_VERSION=\"22.4.0\" -DREDIS_SOCKET_PATH=\"/tmp/redis.sock\" -Dgvm_util_shared_EXPORTS -I/usr/x86_64-pc-linux-gnu/include/glib-2.0 -I/usr/x86_64-pc-linux-gnu/lib/glib-2.0/include -I/usr/x86_64-pc-linux-gnu/include/libxml2 -march=native -O2 -pipe      -std=c11     -Wall     -Wextra     -Werror     -Wpedantic     -Wmissing-prototypes     -Wshadow     -Wsequence-point     -D_ISOC11_SOURCE     -D_DEFAULT_SOURCE -D_FILE_OFFSET_BITS=64 -DLARGEFILE_SOURCE=1 -O3 -DNDEBUG -Wformat -Wformat-security -D_FORTIFY_SOURCE=2 -fstack-protector -fPIC -std=gnu11 -MD -MT util/CMakeFiles/gvm_util_shared.dir/mqtt.c.o -MF CMakeFiles/gvm_util_shared.dir/mqtt.c.o.d -o CMakeFiles/gvm_util_shared.dir/mqtt.c.o -c /var/tmp/paludis/build/net-analyzer-gvm-libs-22.4.0/work/gvm-libs-22.4.0/util/mqtt.c
make[2]: Leaving directory '/var/tmp/paludis/build/net-analyzer-gvm-libs-22.4.0/work/build'
make[1]: Leaving directory '/var/tmp/paludis/build/net-analyzer-gvm-libs-22.4.0/work/build'
In file included from /var/tmp/paludis/build/net-analyzer-gvm-libs-22.4.0/work/gvm-libs-22.4.0/util/mqtt.c:41:
/var/tmp/paludis/build/net-analyzer-gvm-libs-22.4.0/work/gvm-libs-22.4.0/util/mqtt.h:28:10: fatal error: MQTTClient.h: No such file or directory
   28 | #include <MQTTClient.h>
      |          ^~~~~~~~~~~~~~
compilation terminated.
[...]
```

### Expected behavior
Allow to build without optional mqtt support

### Actual behavior
mqtt support contrary to the commen in the CMakeLists.txt https://github.com/greenbone/gvm-libs/blob/stable/util/CMakeLists.txt#L54 is not optional.

### Steps to reproduce

1. download gvm-libs-22.4.0.tar.gz
2. try to build
3. see error

### GVM versions

**gvm-libs:** -

### Environment

**Operating system:**

Exherbo Linux

**Installation method / source:** source-based "packages"

### Logfiles

complete build log: 
[gvm-libs-22.4.0.log](https://github.com/greenbone/gvm-libs/files/9142226/gvm-libs-22.4.0.log)

--- a/INSTALL.md
+++ b/INSTALL.md
@@ -2,7 +2,7 @@ INSTALLATION INSTRUCTIONS FOR GVM-LIBS
 ======================================
 
 Please note: The reference system used by most of the developers is Debian
-GNU/Linux 'Buster' 10. The build might fail on any other system. Also, it is
+GNU/Linux 'Bullseye' 11. The build might fail on any other system. Also, it is
 necessary to install dependent development packages.
 
 Prerequisites for gvm-libs
@@ -29,6 +29,7 @@ Specific development libraries:
 * libnet1 >= 1.1.2.1 (boreas)
 * libpcap
 * libgcrypt
+* libpaho-mqtt >= 1.3.0 (utils)
 
 Prerequisites for building documentation:
 * doxygen
@@ -37,7 +38,7 @@ Prerequisites for building documentation:
 Prerequisites for building tests:
 * [Cgreen](https://cgreen-devs.github.io/#_installing_cgreen) (optional, for building tests)
 
-Install prerequisites on Debian GNU/Linux 'Buster' 10:
+Install prerequisites on Debian GNU/Linux 'Bullseye' 11:
 
     apt-get install \
     cmake \
@@ -50,7 +51,8 @@ Install prerequisites on Debian GNU/Linux 'Buster' 10:
     libhiredis-dev \
     libxml2-dev \
     libpcap-dev \
-    libnet1-dev
+    libnet1-dev \
+    libpaho-mqtt-dev
 
 
 Prerequisites for Optional Features
@@ -70,9 +72,6 @@ Install prerequisites for optional features on Debian GNU/Linux 'Buster' 10:
     apt-get install \
     libldap2-dev \
     libradcli-dev
-
-Prerequisites for MQTT support:
-* libpaho-mqtt-dev >= 1.3.8. This package is currently not available in debian buster stable. Could be installed from source, backports or unstable branch.
 
 Compiling gvm-libs
 ------------------

--- a/util/CMakeLists.txt
+++ b/util/CMakeLists.txt
@@ -51,11 +51,10 @@ pkg_check_modules (REDIS REQUIRED hiredis>=0.10.1)
 pkg_check_modules (LIBXML2 REQUIRED libxml-2.0>=2.0)
 
 # for mqtt
-# Optional for now because lib is currently not in debian buster stable.
 find_library(LIBPAHO paho-mqtt3c)
 message (STATUS "Looking for paho-mqtt3c ... ${LIBPAHO}")
 if (NOT LIBPAHO)
-  message (STATUS "libpaho-mqtt3c is required for MQTTv5 support.")
+  message (SEND_ERROR "libpaho-mqtt3c is required for MQTTv5 support.")
 else (LIBPAHO)
   set (LIBPAHO_LDFLAGS "paho-mqtt3c")
   add_definitions (-DHAVE_MQTT=1)

